### PR TITLE
Added maplike and setlike handling in idlharness.js

### PR DESCRIPTION
**Pull request by ghost**
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/9878)

### CODE REVIEW

**Comment by manikishan** [comment]
issue #8616
I have declared an add_maplike_members function and called that function in a case "maplike" in test_members

File: resources/idlharness.js [hunk 1]
```diff
@@ -2194,7 +2217,10 @@ IdlInterface.prototype.test_members = function()
         case "iterable":
             this.add_iterable_members(member);
             break;
-        // TODO: add setlike and maplike handling.
+        case "maplike":
+            this.add_maplike_members(member);
+            break;
+         // TODO: add setlike and maplike handling.
```
Comment (jgraham) [inline resources/idlharness.js]: This comment is now wrong.

**Review by jgraham — COMMENTED** [review]
I think this looks reasonable, thanks! But I would like @tobie or @foolip to take another look.

File: resources/idlharness.js (hunk 1 above)
Comment (manikishan) [reply resources/idlharness.js]: OK I will change.May I know whether the implementation of maplike is correct?

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2018-03-07 12:01:54
Finished: 2018-03-07 12:02:41


<h2>Failing Jobs</h2>
<ul>

<li>resources_unittest in py27</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25377)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/349859092)

**Comment by manikishan** [comment]
Can I know how to get over the failure case?

**Comment by foolip** [comment]
@manikishan, we've had some trouble with resources_unittest lately, to rule out those problem can you rebase your changes on master and push, so that Travis will run again?

**Comment by foolip** [comment]
Unfortunately, with our current Travis setup it's very hard to know what the consequence of a change like this across all tests is going to be. I'll try running it through Chromium's CQ to see which tests will be affected.

**Comment by manikishan** [comment]
@foolip  Thanks, I will rebase and push it again

**Comment by foolip** [comment]
I'm trying the change in https://chromium-review.googlesource.com/c/chromium/src/+/977961

**Comment by foolip** [comment]
Results are in https://test-results.appspot.com/data/layout_results/linux_chromium_rel_ng/52585/layout-test-results/results.html

Looks like all idlharness.js tests are broken. Can you test your changes with some of the tests locally? When you think it's in good shape, I can try it again.

File: resources/idlharness.js [hunk 2]
```diff
@@ -2067,6 +2067,50 @@ IdlInterface.prototype.add_iterable_members = function(member)
...
            { name: "thisValue", idlType: {idlType: "any"}, optional: true}]}));
 };
+//@}
+IdlInterface.prototype.add_maplike_members = function(member)
+//@{
+{
+    this.members.push(new IdlInterfaceMember(
+        { type: "operation", name: "size", idlType: "maplike", arguments: []}));
+    this.members.push(new IdlInterfaceMember(
+        { type: "operation", name: "entries", idlType: "maplike", arguments []}));
```
Comment (foolip) [inline resources/idlharness.js]: The problem is a missing : here and elsewhere.

Comment (manikishan) [reply resources/idlharness.js]: I will change it soon. Thank you!

**Comment by foolip** [comment]
I'll try it again.

**Comment by foolip** [comment]
Bots now running in https://chromium-review.googlesource.com/c/chromium/src/+/977961

**Comment by foolip** [comment]
Results are in and it looks like only a single test is affected, webaudio/idlharness.https.html:
https://test-results.appspot.com/data/layout_results/linux_chromium_rel_ng/53647/layout-test-results/results.html

That's less than I expected. @manikishan, interfaces/webrtc-pc.idl has "maplike" and interfaces/css-font-loading.idl has "setlike". Can you check the tests involving those to see if the results are what you expected?

**Comment by manikishan** [comment]
@foolip , Sure I will Thank you!

**Comment by manikishan** [comment]
Can I please know which tests I have to run to check the interfaces? @foolip

**Comment by foolip** [comment]
@manikishan, these are the files and how I found them:
```bash
$ git grep -l webrtc-pc.idl
webrtc/interfaces.https.html
$ git grep -l font-loading.idl
css/css-font-loading/idlharness.https.html
```

**Comment by manikishan** [comment]
Yes, I found them too. I was meant to ask how could I check it's implementation. I think I have to run some tests and if yes, what tests should I run? Sorry if I misunderstood anything

**Comment by foolip** [comment]
@manikishan, with your changes, has the total number of tests increased? And are the new tests passing or failing as you would expect them to? I ask because with `iterable` it turned out that there was something a bit wrong: https://github.com/w3c/web-platform-tests/pull/9790

Still haven't figured out exactly how to fix that, but if there are similar issues with maplike and setlike it'd be good to know before they show up as incorrectly failing tests.

**Comment by manikishan** [comment]
Tests in local build tests, or Travis ?

**Comment by manikishan** [comment]
@foolip  I have run the tests in the local repository and found two failures .

1. ```
assert_equals: expected (undefined) undefined but got (function) function "function DOMError() { [native code] }"

isInterfaceNuked/<@http://web-platform.test:8000/dom/historical.html:10:5
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1501:20
test@http://web-platform.test:8000/resources/testharness.js:511:9
isInterfaceNuked@http://web-platform.test:8000/dom/historical.html:9:3
@http://web-platform.test:8000/dom/historical.html:32:1
```

2. ```
assert_equals: expected (undefined) undefined but got (boolean) true

isNukedFromDocument/<@http://web-platform.test:8000/dom/historical.html:38:5
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1501:20
test@http://web-platform.test:8000/resources/testharness.js:511:9
isNukedFromDocument@http://web-platform.test:8000/dom/historical.html:35:3
@http://web-platform.test:8000/dom/historical.html:60:1

```
Does these failures give any hint to us?

**Comment by foolip** [comment]
@manikishan, it looks like you were running dom/historical.html instead of webrtc/interfaces.https.html and css/css-font-loading/idlharness.https.html.

In the affected tests, are the new failing subtests? You can try it in any browser you think supports the maplike and setlike bits of these interfaces. If none do, then just check to see if the failures are as you would expect them to be, and not failures because of some syntax error or similar.

**Comment by manikishan** [comment]
@foolip  I tried to run tests on webrtc/interfaces.https.html  and css/css-font-loading/idlharness.https.html The result was an error stating no tests were run

**Comment by foolip** [comment]
@manikishan did that error happen before your changes as well? If not, you'll need to figure out what in your changes broke the tests. Sorry to say that tooling for detecting regressions doesn't exist yet, so this is all time-consuming and frustrating for everyone :/

**Comment by foolip** [comment]
Closing because the user who sent this has been deleted, and there are conflicts now.

**Pull request closed without merging**